src/hotspot/share/cds/archiveUtils.cpp

```diff
@@ -164,8 +164,7 @@ char* DumpRegion::expand_top_to(char* newtop) {
       // This is just a sanity check and should not appear in any real world usage. This
       // happens only if you allocate more than 2GB of shared objects and would require
       // millions of shared classes.
-      vm_exit_during_initialization("Out of memory in the CDS archive",
-                                    "Please reduce the number of shared classes.");
+      MetaspaceShared::unrecoverable_writing_error("Out of memory in the CDS archive\nPlease reduce the number of shared classes.");
```

Comment:
This should probably be two log_error() messages rather than a two line message with a \n in it.  Or remove the \n and add a period and blank, to fix the original message.  There aren't typically multiline comments with \n in the jvm.